superset-frontend/packages/superset-ui-chart-controls/src/utils/savedMetricsTypeCheck.ts

```diff
@@ -0,0 +1,30 @@
+/* eslint-disable camelcase */
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { QueryResponse } from '@superset-ui/core';
+import { Dataset } from '../types';
+import { DEFAULT_METRICS } from '..';
+
+export const savedMetricsTypeCheck = (
```

Comment:
to be more specific, I would say that this function is either fetching/defining or creating metrics. The type checking is just a side effect, if you wanted to rename this to something that explains what it does a little more succinctly.

Reply:
Good idea! I changed it to `defineSavedMetrics` in [`this commit`](https://github.com/apache/superset/pull/19855/commits/cea65ce3bf31f7565bf0ddb75cebffbcb89ac1e2).